### PR TITLE
Issue 482 misc cleanup

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

**Comment by mortenjc** [comment]
1) Minor stuff - remove duplicate and unused stats
2) Move 'generic' udpgen away from common/readout/vmm3 as it no longer is vmm3 specific
3) Small changes to ReadoutGeneratorBase to support different sizes of readouts.
4) Remove old and create new loki generator (makes a better but still pretty bad detector image)
5) Move Ring, FEN and DataSize into the Loki readouts since we now always only expect one readout per data header

File: src/modules/cspec/generators/CMakeLists.txt [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 set(udpgen_vmm3_cspec_INC
   ${ESS_SOURCE_DIR}/common/readout/vmm3/VMM3Parser.h
-  ${ESS_SOURCE_DIR}/common/readout/vmm3/generators/ReadoutGeneratorBase.h
+  ${ESS_SOURCE_DIR}//generators/essudpgen/ReadoutGeneratorBase.h
```
Comment (mortenjc) [inline src/modules/cspec/generators/CMakeLists.txt:7]: Check double //

File: src/modules/loki/configs/STFCTestIII.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+{
+  "Comment" : "March 2022 test - Ring 0 has 16 FENs, part of bank 0",
+
+  "Detector": "LoKI4x2",
```
Comment (mortenjc) [inline src/modules/loki/configs/STFCTestIII.json:5]: wrong name

Comment (mortenjc) [reply src/modules/loki/configs/STFCTestIII.json:5]: require correct name for Detector

File: src/modules/loki/test/CombinedParserTest.cpp [hunk 3]
```diff
@@ -100,8 +100,8 @@ TEST_F(CombinedParserTest, ParseUDPPacket) {
   for (auto & Section : LokiParser.Result) {
     printf("Ring %u, FEN %u\n", Section.RingId, Section.FENId);
     printf("time (%u, %u), SeqNum %u, Tube %u, A %u, B %u, C %u, D %u\n",
-            Section.Data.TimeHigh, Section.Data.TimeLow, Section.Data.DataSeqNum, Section.Data.TubeId,
-            Section.Data.AmpA, Section.Data.AmpB, Section.Data.AmpC, Section.Data.AmpD);
```
Comment (mortenjc) [inline src/modules/loki/test/CombinedParserTest.cpp:104]: Section -> Data

**Review by JennyLouise — APPROVED** [review]

File: src/modules/cspec/generators/CMakeLists.txt (hunk 1 above)
Comment (mortenjc) [reply src/modules/cspec/generators/CMakeLists.txt:7]: done

File: src/modules/loki/configs/STFCTestIII.json (hunk 2 above)
Comment (mortenjc) [reply src/modules/loki/configs/STFCTestIII.json:5]: Changed name to LoKI and added check that throws a runtime error if wrong

File: src/modules/loki/test/CombinedParserTest.cpp (hunk 3 above)
Comment (mortenjc) [reply src/modules/loki/test/CombinedParserTest.cpp:104]: done

File: src/modules/loki/test/CMakeLists.txt [hunk 4]
```diff
@@ -80,15 +80,15 @@ create_test_executable(LokiDataParserTest)
...
-  ../readout/DataParser.cpp
-  CombinedParserTest.cpp
-  ReadoutGenerator.cpp)
-create_test_executable(CombinedParserTest)
+#set(CombinedParserTest_INC
+#  ../readout/DataParser.h
+#  ${ESS_SOURCE_DIR}/common/readout/ess/Parser.h
+#  ${ESS_SOURCE_DIR}/common/testutils/DataFuzzer.h
+#  )
+#set(CombinedParserTest_SRC
```
Comment (mortenjc) [inline src/modules/loki/test/CMakeLists.txt:88]: will reinstate this code

**Pull request merged**